### PR TITLE
[QnA] Fix QnAMakerRecognizerTests unit tests

### PR DESCRIPTION
**Pull request by Batta32**
Fixes #

## Specific Changes

  - 
  -
  -

### CODE REVIEW

File: libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/HttpRequestUtils.java [hunk 1]
```diff
@@ -43,29 +42,37 @@
...
-                            .append(response.code()).append(".").toString();
-                    throw new Exception(message);
-                }
-            } catch (Exception e) {
-                LoggerFactory.getLogger(HttpRequestUtils.class).error("findPackages", e);
-                throw new CompletionException(e);
-            }
+    private Request buildRequest(String requestUrl, String endpointKey, RequestBody body) {
+        HttpUrl.Builder httpBuilder = HttpUrl.parse(requestUrl).newBuilder();
+        Request.Builder requestBuilder = new Request.Builder()
```
Comment (fran893) [inline libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/HttpRequestUtils.java:68]: I think here we need to add the body into the request

File: libraries/bot-ai-qna/src/test/java/com/microsoft/bot/ai/qna/QnAMakerRecognizerTests.java [hunk 2]
```diff
@@ -73,24 +75,47 @@ public void noTextNoAnswer() {
...
         MockWebServer mockWebServer = new MockWebServer();
         try {
-            this.initializeMockServer(mockWebServer, "QnaMaker_ReturnsAnswer.json", "/knowledgebases/");
+            // Get Oracle file
+            String content = readFileContent("QnaMaker_ReturnsNoAnswer.json");
+            //Extract V3 response
+            ObjectMapper mapper = new ObjectMapper();
+            JsonNode response = mapper.readTree(content);
+            // Set mock response in MockWebServer
+            String url = "/qnamaker/knowledgebases/";
```
Comment (fran893) [inline libraries/bot-ai-qna/src/test/java/com/microsoft/bot/ai/qna/QnAMakerRecognizerTests.java:84]: This variable is used several time, can we make it global and final?

File: libraries/bot-ai-qna/src/test/java/com/microsoft/bot/ai/qna/QnAMakerRecognizerTests.java [hunk 3]
```diff
@@ -99,22 +124,46 @@ public void noAnswer() {
...
+            validateAnswers(result);
+            Assert.assertEquals(result.getIntents().get("None"), null);
+            Assert.assertNotEquals(result.getIntents().get("QnAMatch"), null);
+        } catch (Exception e) {
+            fail();
         } finally {
             try {
                 mockWebServer.shutdown();
             } catch (IOException e) {
-                LoggerFactory.getLogger(QnAMakerRecognizerTests.class).error(e.getMessage());
```
Comment (fran893) [inline libraries/bot-ai-qna/src/test/java/com/microsoft/bot/ai/qna/QnAMakerRecognizerTests.java:117]: Why do you remove this in all the catch statement?

**Review by fran893 — CHANGES_REQUESTED** [review]

File: libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/HttpRequestUtils.java (hunk 1 above)
Comment (Batta32) [reply libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/HttpRequestUtils.java:68]: Done

File: libraries/bot-ai-qna/src/test/java/com/microsoft/bot/ai/qna/QnAMakerRecognizerTests.java [hunk 4]
```diff
@@ -73,24 +75,47 @@ public void noTextNoAnswer() {
...
+            // Get Oracle file
+            String content = readFileContent("QnaMaker_ReturnsNoAnswer.json");
+            //Extract V3 response
+            ObjectMapper mapper = new ObjectMapper();
+            JsonNode response = mapper.readTree(content);
+            // Set mock response in MockWebServer
+            String url = "/qnamaker/knowledgebases/";
+            String endpoint = "";
+            if (this.mockQnAResponse) {
+                endpoint = String.format(
```
[the remark this answers is not in the document]
Comment (Batta32) [reply libraries/bot-ai-qna/src/test/java/com/microsoft/bot/ai/qna/QnAMakerRecognizerTests.java:87]: Nope, I'll change it

File: libraries/bot-ai-qna/src/test/java/com/microsoft/bot/ai/qna/QnAMakerRecognizerTests.java [hunk 5]
```diff
@@ -99,22 +124,46 @@ public void noAnswer() {
...
+            // Get Oracle file
+            String content = readFileContent("QnaMaker_ReturnsAnswer.json");
+            //Extract V3 response
+            ObjectMapper mapper = new ObjectMapper();
+            JsonNode response = mapper.readTree(content);
+            // Set mock response in MockWebServer
+            String url = "/qnamaker/knowledgebases/";
+            String endpoint = "";
+            if (this.mockQnAResponse) {
+                endpoint = String.format(
```
[the remark this answers is not in the document]
Comment (Batta32) [reply libraries/bot-ai-qna/src/test/java/com/microsoft/bot/ai/qna/QnAMakerRecognizerTests.java:136]: Same

File: libraries/bot-ai-qna/src/test/java/com/microsoft/bot/ai/qna/QnAMakerRecognizerTests.java [hunk 6]
```diff
@@ -123,22 +172,46 @@ public void returnAnswers() {
...
+            // Get Oracle file
+            String content = readFileContent("QnaMaker_TopNAnswer.json");
+            //Extract V3 response
+            ObjectMapper mapper = new ObjectMapper();
+            JsonNode response = mapper.readTree(content);
+            // Set mock response in MockWebServer
+            String url = "/qnamaker/knowledgebases/";
+            String endpoint = "";
+            if (this.mockQnAResponse) {
+                endpoint = String.format(
```
[the remark this answers is not in the document]
Comment (Batta32) [reply libraries/bot-ai-qna/src/test/java/com/microsoft/bot/ai/qna/QnAMakerRecognizerTests.java:184]: Same

File: libraries/bot-ai-qna/src/test/java/com/microsoft/bot/ai/qna/QnAMakerRecognizerTests.java [hunk 7]
```diff
@@ -147,77 +220,72 @@ public void topNAnswers() {
...
+            String content = readFileContent("QnaMaker_ReturnsAnswerWithIntent.json");
+            //Extract V3 response
+            ObjectMapper mapper = new ObjectMapper();
+            JsonNode response = mapper.readTree(content);
+            // Set mock response in MockWebServer
+            String url = "/qnamaker/knowledgebases/";
+            String endpoint = "";
+            if (this.mockQnAResponse) {
+                endpoint = String.format(
+                    "%s:%s",
```
[the remark this answers is not in the document]
Comment (Batta32) [reply libraries/bot-ai-qna/src/test/java/com/microsoft/bot/ai/qna/QnAMakerRecognizerTests.java:233]: Same

File: libraries/bot-ai-qna/src/test/java/com/microsoft/bot/ai/qna/QnAMakerRecognizerTests.java [hunk 8]
```diff
@@ -6,19 +6,19 @@
...
 import com.fasterxml.jackson.databind.ObjectMapper;
-import com.microsoft.bot.builder.MessageFactory;
 import com.microsoft.bot.builder.RecognizerResult;
 import com.microsoft.bot.builder.TurnContext;
 import com.microsoft.bot.builder.TurnContextImpl;
 import com.microsoft.bot.builder.adapters.TestAdapter;
-import com.microsoft.bot.dialogs.DialogContext;
-import com.microsoft.bot.dialogs.DialogSet;
-import com.microsoft.bot.dialogs.DialogState;
+import com.microsoft.bot.dialogs.*;
```
[the remark this answers is not in the document]
Comment (Batta32) [reply libraries/bot-ai-qna/src/test/java/com/microsoft/bot/ai/qna/QnAMakerRecognizerTests.java:18]: Nope, I'll change it

**Review by fran893 — APPROVED** [review]

**Review by FedericoBernal — APPROVED** [review]
LGTM

**Pull request merged**